ui/app/src/state/reducers/pathNavigator.ts

```diff
@@ -164,6 +170,10 @@ const reducer = createReducer<LookupTable<PathNavigatorStateProps>>(
           itemsInPath: children.map((item) => item.path),
           levelDescriptor: children.levelDescriptor?.path ?? null,
           breadcrumb: getIndividualPaths(withoutIndex(currentPath), withoutIndex(rootPath)),
+          leaves: [
+            ...state[id].leaves,
+            ...children.filter((item) => item.childrenCount === 0).map((item) => item.path)
```

Comment:
Create a local util fn that receives children and returns an array of leaf paths and use it across all these instances